docs/contributing.md

```diff
@@ -1 +1,185 @@
-# Contributing
\ No newline at end of file
+# Contributing to RE-UE4SS
+
+Thank you for taking the time to contribute!
+
+All types of contributions are encouraged and valued. This guide outlines the process and guidelines for contributing to the project.
+
+## Table of Contents
+
+- [Code of Conduct](#code-of-conduct)
+- [I Have a Question](#i-have-a-question)
+- [Reporting Issues](#reporting-issues)
+- [Pull Request Guidelines](#pull-request-guidelines)
+- [Development Workflow](#development-workflow)
+- [Code Style Guidelines](#code-style-guidelines)
+- [Commit Message Guidelines](#commit-message-guidelines)
+- [Upgrade Guidelines](#upgrade-guidelines)
+- [License](#license)
+
+## Code of Conduct
+
+This project and everyone participating in it is governed by the
+[RE-UE4SS Code of Conduct](https://github.com/UE4SS-RE/RE-UE4SS/blob/main/CODE_OF_CONDUCT.md).
+By participating, you are expected to uphold this code. Please report unacceptable behavior
+to <contact@ue4ss.com>.
+
+## I Have a Question
+
+Before you ask a question, it is best to search for existing [Issues](https://github.com/UE4SS-RE/RE-UE4SS/issues) and read the available [Documentation](https://docs.ue4ss.com/).
+
+If you still need clarification:
+- Open an [Issue](https://github.com/UE4SS-RE/RE-UE4SS/issues/new)
+- Provide as much context as you can about what you're experiencing
+- Provide project and platform versions (OS, compiler, etc.)
+
+## Reporting Issues
+
+When reporting issues, please follow our issue template:
+
+```
+**Branch or Release**
+[Specify which branch or release version you're using]
+
+**Game and Engine Version**
+[Specify the game and UE version]
+
+**Describe the bug**
+[A clear description of the bug]
+
+**Mods directory**
+[Relevant mods directory structure if applicable]
+
+**To reproduce**
+[Step-by-step reproduction steps]
+
+**Expected behavior**
+[What you expected to happen]
+
+**Screenshots, UE4SS Log, and .dmp file**
+[Attach relevant files]
+
+**Desktop (please complete the following information):**
+* OS: [e.g. Win10, Win11]
+
+**Additional context**
+[Any other relevant information]
+```
+
+For good bug reports:
+- Explain the behavior you expect and the actual behavior
+- Provide clear reproduction steps that someone else can follow
+- Isolate the problem and create a reduced test case if possible
+- Verify if the issue occurs with the latest version of RE-UE4SS
+
+> **Security Issues**: Never report security-related issues, vulnerabilities or bugs including sensitive information to the issue tracker. Instead, please email sensitive bugs to <security@ue4ss.com>.
+
+## Pull Request Guidelines
+
+### General Requirements
+
+1. Before submitting a PR, please ensure your changes have been tested on:
```

Comment:
Perhaps some mention that this is only for actual UE4SS/UEPsuedo/dependencies code changes, though not sure how to word that and to be honest it's common sense that you wouldn't need to do that for a docs-only change etc.